src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -0,0 +1,2835 @@
+```
+
+If the type declaration is omitted,
+the runtime type will be `Cache<dynamic>`
+and both calls to `setByKey` are valid.
+
+### Restricting generics
+
+You can use generics to restrict your code to
+a family of types using `extends`. This ensures
+that your class is instantiated with a generic type
+that extends a specific type:
+
+_Dart_
+```dart
+class NumberManager<T extends num> {
+   ...
+}
+// Valid.
+var manager = NumberManager<int>(); 
+var manager = NumberManager<double>(); 
+// Invalid, String nor its parent classes extend num.
+var manager = NumberManager<String>(); 
+```
+
+### Generics in literals
+
+`Map`-, `Set`- and `List` literals can explicitly
+declare generic types, which is useful when the
+type isn’t inferred or is incorrectly inferred. 
+
+For example, the `List` class has a generic definition:
+`class List<E>`. Generic type `E` refers to the type of
+the list’s contents. Normally, this type is automatically inferred,
+which is used in some `List` class’s member types.
+(For example, its first getter returns a value of type `E`.)
+When defining a `List` literal,
+you can explicitly define the generic type as follows:
+
+_Dart_
+```dart
+// Automatic type inference
+var objList = [5, 2.0]; // Type: List<num>
+// Explicit type definition:
+var objList = <Object>[5, 2.0]; // Type: List<Object>
+// Sets work identically:
+var objSet = <Object>{5, 2.0};
+```
+
+This is also true for `Map`s,
+which also define their key and value types
+using generics (`class Map<K, V>`):
+
+_Dart_
+```dart
+// Automatic type inference
+var map = {
+  'foo': 'bar'
+}; // Type: Map<String, String>
+// Explicit type definition:
+var map = <String, Object>{
+  'foo': 'bar'
+}; // Type: Map<String, Object>
+```
+
+## Doc comments
+
+Regular comments work the same in Dart as they do
+in JavaScript. Using `//` comments out everything beyond
+it for the remaining line, and you can use `/* ... */`
+to block comments spanning multiple lines.
+
+In addition to regular comments,
+Dart also has [doc comments][] that work in tandem
+with [dartdoc][]: a first party tool that generates
+HTML documentation for Dart packages.
+It's considered best practice to place doc comments
+above all declarations for public members.
+
+Define a doc comment by using three forward slashes
+instead of two (`///`):
+
+_Dart_
+```dart
+/// The number of characters in this chunk when unsplit.
+int get length => ...
+```
+
+[dartdoc]: /tools/dart-doc
+[doc comments]: /guides/language/effective-dart/documentation#doc-comments
+
+## Next steps
+
+This guide has introduced you to the major differences
+between Dart and JavaScript. At this point,
+you might consider moving to the general documentation
+for Dart or Flutter (an open-source framework that
+uses Dart for building beautiful, natively compiled,
+multi-platform applications from a single codebase),
+where you'll find in-depth information about the
+language and practical ways of getting started. 
+
```

Comment:
Maybe we can link to some good resources or next steps here?

Some possibilities:

- [Language tour](https://dart.dev/guides/language/language-tour) to learn more about the language
- [Library tour](https://dart.dev/guides/libraries/library-tour) to learn more about Dart’s core libraries
- [Dart codelabs](https://dart.dev/codelabs) for hands on experience learning Dart or various topics
- [Effective Dart](https://dart.dev/guides/language/effective-dart) to learn about common conventions and guidelines when writing Dart code